lib/galaxy/jobs/command_factory.py

```diff
@@ -294,11 +294,10 @@ def __copy_if_exists_command(work_dir_output):
 
 class CommandsBuilder:
     def __init__(self, initial_command=""):
-        # Remove trailing semi-colon so we can start hacking up this command.
         # TODO: Refactor to compose a list and join with ';', would be more clean.
         self.raw_command = initial_command
         initial_command = util.unicodify(initial_command or "")
-        commands = initial_command.rstrip("; ")
+        commands = initial_command.rstrip(" ")
```

Comment:
Have you considered the suggestion in the TODO ? (Refactor to compose a list and join with ';', would be more clean.)

Reply:
I thought about it. But I'm not sure how to treat https://github.com/galaxyproject/galaxy/blob/32f7d3bd1a65fb10b2b95043c0450e05a7289cc9/lib/galaxy/jobs/command_factory.py#L324 which does not join with `;`

Anyway this seems independent of the question if the trailing `;` can be of any harm.

Reply:
OK. I tried in https://github.com/galaxyproject/galaxy/pull/14713/commits/5ffa6943cd01e237b72def15ef020376b8b33d3c

Do you see a possibility to use the correct trap from the start? Instead of the `replace`?